### ISSUE TITLE
Release 7.2.0

### ISSUE THREAD

**Issue by rahulporuri** [enthought/traitsui#1569]
[Issues need to be created and work needs to be done to raise `DeprecationWarning`s for code which is documented as deprecated but doesnt actually raise said warnings]

Target date:

~Release Candidate by YYYY-MM-DD~

Release by 2021-05-11

Release blockers
----------------
- [ ] (add blocking issues/PRs here)

Pre-release
---
- [x] Verify that all relevant PRs have been merged to master.
- [x] Create a PR against master to bump version number, merge that PR (#1651)
- [x] From the commit just before bumping the version, create a new branch `maint/<release version number>`
- [x] Update changelog and open PR targeting a new `maint/<release version number>` branch (done on master first, see #1611)
- [x] Update `ci-src-requirements.txt` if needed
- [x] Check MANIFEST and requirements are still up to date.
- [x] Backport PRs that have been merged to master to the maintenance branch. Use the "need backport ..." tag if there is one (but don't rely 100% on it) (#1661)
- [x] Verify that no other open issue needs to be addressed before the release.
- [x] Test against other ETS packages and other ETS-using projects
- [x] Check MANIFEST, requirements, changelog are still up to date.
- [x] Test building the documentation

Release
-------
- [x] Create branch `release/<release version number>` from `maint/<release version number>` branch.
  - [x] Set release to `<release version number>`, and set `IS_RELEASED` is true; commit
  - [x] Install from source distribution and run tests again
  - [x] Open a PR against `maint/<release version number>` with this being the last commit so that CI is built on the release commit (#1662)
  - [x] Once CI is done building merge PR
  - [x] Bump the micro version number i.e. `<release version number + 0.0.1>` and set `IS_RELEASED` to false; commit.
  - [x] Open and then merge a separate PR against `maint/<release version number>` (#1663)
- [x] From the commit at which `IS_RELEASED` is true and version is `<release version number>`, tag (annotated!) `git tag -a -m "Release <release version number>" <release version number>`
- [x] Push the tag `git push origin <release version number>`
- [x] Make PR targeting `gh-pages` branch: Generate documentation and copy the content to the branch. Verify that the resulting index.html looks good. (#1664)
- [x] Upload to PyPI
- [x] Test the PyPI package

Post-release
-------------
- [x] Package update for `enthought/free` repository (for EDM)
- [x] Backport release note and change log to master, and possibly `maint/<release version number>` branch. (#1665)
- [x] Update GitHub Release pages https://github.com/enthought/traitsui/releases
- [x] Announcement (e.g. Google Group)

**Comment by aaronayres35** [comment]
(relatively recent) Open/WIP PRs we might want to get pushed through before the release: 
- [ ] #1568 
- [ ] #1566 
- [ ] #1560 
- [ ] #1559 
- [ ] #1549 
- [ ] #1544
- [ ] #1520 
- [ ] #1517 
- [ ] #1499
